api/test/unit/routes/policies.test.js

```diff
@@ -2,91 +2,51 @@
-		"operator" : ">=",
-		"cool_down_secs" : 300,
-		"adjustment" : "+1"
-	} ],
-	"schedules" : {
-		"timezone" : "(GMT+08: 00)Asia/Shanghai",
-		"recurring_schedule" : [ {
-			"start_time" : "00: 00",
-			"end_time" : "08: 00",
-			"days_of_week" : [ 1, 2, 3 ],
-			"days_of_month" : [ 1, 28 ],
-			"instance_min_count" : 2,
-			"instance_max_count" : 10
-		} ],
-		"specific_date" : [ {
-			"start_date" : "2015-06-19",
-			"start_time" : "00: 00",
-			"end_date" : "2015-06-19",
-			"end_time" : "23: 59",
-			"instance_min_count" : 15,
-			"instance_max_count" : 40
-		} ]
-	}
-};
 
-describe('Routing Policy', function() {
-  before(function() {
-    this.models = require('../../../lib/models')();
-    this.Policy = this.models.policy_json;
-    this.Policy.sequelize.sync().then(function() {
-      logger.info('creating the table structure');
-    });
-  });
-  describe('Policy Creation', function() {
-    it('should create a policy for app id 12345', function(done) {
-      request(app)
-      .put('/v1/apps/12345/policy')
-      .send(fakePolicy)
-      .end(function(error,result) {
-        if (error) {
-          expect(error).to.not.be.null;
-          throw error;
-        }
-        expect(result.statusCode).to.equal(200);
+describe('Routing Policy Creation', function() {
+  var fakePolicy;
+  before(function(done) {
+    this.policy = require('../../../lib/models')().policy_json;
+    var _this = this;
+    this.policy.sequelize.sync().then(function(success) {
+      _this.policy.truncate().then(function(result) {
+        fakePolicy = JSON.parse(fs.readFileSync(__dirname+'/../fakePolicy.json', 'utf8'));
         done();
-      });
-    
+      });  
+    }, function(error) {
+      logger.error('Failed to setup database for test');
+      done(error);
     });
-
-    it('should fail to create another policy for app id 12345', function(done) {
-      request(app)
-      .put('/v1/apps/12345/policy')
-      .send(fakePolicy)
-      .end(function(error,result) {
-        if (error) {
-          expect(error).to.not.be.null;
-          throw error;
-        }
-        expect(result.statusCode).to.equal(400);
-        done();
-      });
+  });
+  it('should create a policy for app id 12345', function(done) {
+    request(app)
+    .put('/v1/apps/12345/policy')
+    .send(fakePolicy)
+    .end(function(error,result) {
+      expect(result.statusCode).to.equal(201);
+      expect(result.body.success).to.equal(true);
+      expect(result.body.error).to.be.null;
+      expect(result.body.result.policy_json).eql(fakePolicy);
+      done();
     });
   });
 
-  after(function() {
-    this.Policy.truncate().then(function(result) {
-      logger.warn('Truncating the test db before test run !!!');
-    });  
+  it('should fail to create another policy for app id 12345', function(done) {
+    request(app)
+    .put('/v1/apps/12345/policy')
+    .send(fakePolicy)
+    .end(function(error,result) {
+      expect(result.statusCode).to.equal(400);
```

Comment:
Yes. It could be 409 but then, once we have "update-policy" in place, this use case would be redundant. An existing app id would be updated in that case.